app/src/cpp/cfc-cpp/MultiThreadedDecoder.h

```diff
@@ -87,7 +96,9 @@ inline int MultiThreadedDecoder::do_extract(const cv::Mat& mat, cv::Mat& img)
 
 inline bool MultiThreadedDecoder::add(cv::Mat mat)
 {
-	return _pool.try_execute( [&, mat] () {
+    ++count;
+    bool legacy_mode = _modeVal == 4 or (_modeVal == 0 and count%2 == 0);
```

Comment:
Crucially, we do this count -> legacy_mode calculation on the *main* thread, not on the background threads. :no_mouth: